components/Graphic.jsx

```diff
@@ -68,11 +73,13 @@ var Graphic = React.createClass({
 
   componentDidMount: function() {
     var cvs = this.refs.canvas;
-    cvs.width = this.defaultWidth;
-    cvs.height = this.defaultHeight;
+    var dpr = window.devicePixelRatio || 1;
```

Comment:
It's nicer not to have to repeat this evaluation in four different places, so let's instead give the Graphic class a function to retrieve this value:
```
getPixelRatio() {
  return window.devicePixelRatio || 1;
}
```
and then `var dpr = this.getPixelRatio();` in the rest of the code